src/parser/Parser.cpp

```diff
@@ -1,122 +1,115 @@
 #include <Parser.h>
 
-Parser::Parser(Lexer lexer)
-    : m_lexer(std::move(lexer))
-    , m_currentToken(m_lexer.getToken())
+Parser::Parser(const Lexer lexer)
+    : m_lexer(lexer)
 {
+    m_currenttoken = m_lexer.getToken();
+}
 
+Token Parser::eattoken(TokenType t) {
+    if (m_currenttoken.type != t) {
+        puts("Unexpected token type.");
+        exit(1);
+    }
+    m_currenttoken = m_lexer.getToken();
 }
 
-void Parser::buildTree() {
-    parseStatement();
+std::unique_ptr<AST::Stmt> Parser::parse() {
+    return parseStmt();
 }
 
-std::unique_ptr<AST::Stmt> Parser::parseStatement() {
-    std::unique_ptr<AST::Stmt> statement;
-    Token identifier;
-    switch(m_currentToken.type) {
-        case TokenType::Def:
-            statement = parseFunction();
-        case TokenType::Return:
-            statement = parseReturn();
-        case TokenType::Identifier:
-            identifier = m_currentToken;
-            getNextToken();
-            switch(m_currentToken.type) {
-                case TokenType::LeftParen:
-                    getNextToken();
-            }
-            break;
-
-        // todo: other statements
-
-        default:
-            statement = parseExpression();
+std::unique_ptr<AST::Stmt> Parser::parseStmt(){
+    if (m_currenttoken.type == TokenType::If) {
+        eattoken(TokenType::If);
+        std::unique_ptr<AST::Expr> cond = parseExpr();
+        std::unique_ptr<AST::BlockStmt> body = parseBlockStmt();
+
+        return std::unique_ptr<AST::IfStmt>(new AST::IfStmt(std::move(cond), std::move(body), nullptr));
+        // TODO handle else case
+    }
+
+    if (m_currenttoken.type == TokenType::While) {
+        eattoken(TokenType::While);
+        std::unique_ptr<AST::Expr> cond = parseExpr();
+        std::unique_ptr<AST::BlockStmt> body = parseBlockStmt();
+
+        return std::unique_ptr<AST::WhileStmt>(new AST::WhileStmt(std::move(cond), std::move(body)));
+    }
+
+    if (m_currenttoken.type == TokenType::Next) {
+        eattoken(TokenType ::Next);
+        return std::unique_ptr<AST::NextStmt>(new AST::NextStmt());
+    }
+
+    if (m_currenttoken.type == TokenType::Break) {
+        eattoken(TokenType::Break);
+        return std::unique_ptr<AST::NextStmt>(new AST::BreakStmt());
```

Comment:
Type here should be `BreakStmt` not `NextStmt` (that's causing the build to fail).